### ISSUE TITLE
Removing max_steps in pretrain

### ISSUE THREAD

**Issue by rasbt** [Lightning-AI/litgpt#1098]
As commented in `args.py`:

>     # TODO: `pretrain` is the only script using `max_tokens` explicitly. replace it with epoch_size*epochs?

I also wonder if `max_steps` is maybe sufficient here? I do think max tokens could be useful though if we want to reproduce certain models. E.g., I am thinking of the TinyLlama models. For now, we can maybe keep it, but something to discuss.

**Comment by awaelchli** [comment]
I'd like to keep `max_tokens` for pretraining. It is not intuitive and error prone to calculate max steps if you know max tokens. Also in LLM training max steps are somewhat irrelevant because the batch sizes are so large and scale up and down a lot based on the model size, number of GPUs/machines, that the number of steps just aren't an intuitive metric in my opinion.

**Comment by awaelchli** [comment]
In `pretrain.py`, max_steps is currently "unsupported":
https://github.com/Lightning-AI/litgpt/blob/f951f9334610da35c7ecaa7e26e7ba3ac2504dab/litgpt/pretrain.py#L381-L383

**Comment by rasbt** [comment]
This sounds reasonable. And the `max_steps` being unsupported sounds good!
